.mvn/gradle-enterprise.xml

```diff
@@ -0,0 +1,54 @@
+<gradleEnterprise>
+  <server>
+    <url>https://ge.apache.org</url>
+    <allowUntrusted>false</allowUntrusted>
+    <!--
+    credentials provided by jenkins exposing GRADLE_ENTERPRISE_ACCESS_KEY
+    apache members can create their own on the server
+    -->
+  </server>
+  <!-- options are documented at https://docs.gradle.com/enterprise/maven-extension/#configuring_background_uploading -->
+  <buildScan>
+    <backgroundBuildScanUpload>#{isFalse(env['CI'])}</backgroundBuildScanUpload>
+    <!-- Always publish build scans for CI, only if requested locally -->
+    <publish>#{env['CI'] == null ? 'ON_DEMAND' : 'ALWAYS'}</publish>
+    <capture>
+      <goalInputFiles>false</goalInputFiles> <!-- To be enabled locally when debugging goal caching or if we want to enable predictive test selection -->
```

Comment:
I would recommend enabling `goalInputFiles` capture for all builds. This property creates hashes for all file inputs, but does not actually store the files themselves for storage size and privacy reasons. This allows you to troubleshoot builds more effectively, as you may not enable this on a build and then later want this metadata.

Reply:
The [documentation](https://docs.gradle.com/enterprise/maven-extension/#capturing_goal_input_files) for this property starts with 
> Capturing goal input files increases the amount of data transmitted to the build scan server at the end of the build. If the network connection to the build scan server is poor, it may increase the time required to transmit. Additionally, it may also increase the data storage requirements for the build scan server.

So i intended to enable manually for debugging caching misses but keep it disabled overall at least until we decide to enable/try predictive test selection. 
while I'm not too worried about the time to transmit the build scan I was worried about the resources of the apache instance.

Reply:
I have seen the entry in apache INFRA and followed your suggestion